### ISSUE TITLE
Add Error Handling to `ModuleRecoveryAlert` and show error message to the user.

### ISSUE THREAD

**Issue by kuasha420** [google/site-kit-wp#5318]
## Feature Description

Currently, when one or more modules fails to recover for any reason, the backend spits out error(s) but we do not show it anywhere to the user and the failure happens silently. We should add the error message to the UI so user are not confused. 

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* 

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by eclarke1** [comment]
Move this from Triage to AC as it will be a nice to have for Dashboard Sharing and don't want it to get lost in Triage

**Comment by techanvil** [comment]
I've added some provisional ACs, but these will need to be signed off - cc @aaemnnosttv @felixarntz

**Comment by aaemnnosttv** [comment]
IB ✅

**Comment by wpdarren** [comment]
### QA Update: ⚠️ 

@nfmohit the AC mentions that the colour for the error message should be `#d53e36` but when I inspect the test it is appearing as `#ac4220` This could be related to GM2+ styling change, but would like to make sure. 

Thoughts? 

![image](https://user-images.githubusercontent.com/73545194/180939969-16029e13-7495-40db-8b02-669510b4514c.png)

**Verified:**

- In the `ModuleRecoveryAlert` component, when one or more modules fail to recover the error message is displayed.
- When there is only a single module to recover which fails, the message format is as per the AC. 
- When there are multiple module errors, the message format is as per the AC. 

<details>
 <summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/73545194/180940587-8ab61a75-0de6-403b-bd27-6b372c0036e5.png)
![image](https://user-images.githubusercontent.com/73545194/180940553-185d0a57-028b-4deb-89fb-bd5372ebb656.png)

</details>

**Comment by nfmohit** [comment]
@wpdarren You're right. After GM2+, the error styles have been updated globally and have been reflected here as well.

**Comment by wpdarren** [comment]
### QA Update: ✅ 

**Verified:**

- In the `ModuleRecoveryAlert` component, when one or more modules fail to recover the error message is displayed.
- When there is only a single module to recover which fails, the message format is as per the AC. 
- When there are multiple module errors, the message format is as per the AC. 

<details>
 <summary>Screenshots</summary>

![image](https://user-images.githubusercontent.com/73545194/180940587-8ab61a75-0de6-403b-bd27-6b372c0036e5.png)
![image](https://user-images.githubusercontent.com/73545194/180940553-185d0a57-028b-4deb-89fb-bd5372ebb656.png)

</details>
